### PR TITLE
[AIR] Move Constants from tune/results.py to air/constants.py

### PR DESCRIPTION
**Pull request by woshiyyya**
This PR aims to migrate some of the constants from `ray/tune/result.py` to `ray/air/constants.py`.

## Why are these changes needed?

Currently we have result-related constants stored in `ray/tune/result.py`, However, our Result object is defined in `ray/air/result.py`. When we are trying to import constants from `ray/tune/result.py` in `ray/air/result.py`, there will be a cyclic import error:
```
import Result -> import ray.tune.result -> ray.tune.init -> import ResultGrid -> import Result
```

This PR only moves the following constants and changed the import path in all affected files accordingly. In the future, we will gradually move the rest constants along with the module class.
```python
# The timestamp of when the result is generated.
# Default to when the result is processed by tune.
TIMESTAMP = "timestamp"

# (Auto-filled) Time in seconds this iteration took to run.
# This may be overridden to override the system-computed time difference.
TIME_THIS_ITER_S = "time_this_iter_s"

# (Auto-filled) The index of this training iteration.
TRAINING_ITERATION = "training_iteration"

# File that stores parameters of the trial.
EXPR_PARAM_FILE = "params.json"

# Pickle File that stores parameters of the trial.
EXPR_PARAM_PICKLE_FILE = "params.pkl"

# File that stores the progress of the trial.
EXPR_PROGRESS_FILE = "progress.csv"

# File that stores results of the trial.
EXPR_RESULT_FILE = "result.json"

# File that stores the pickled error file
EXPR_ERROR_PICKLE_FILE = "error.pkl"

# File that stores the error file
EXPR_ERROR_FILE = "error.txt"
```

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/constants.py [hunk 1]
```diff
@@ -18,13 +18,6 @@
 )
 
 # Autofilled session.report() metrics. Keys should be consistent with Tune.
-# The train provided `TIME_THIS_ITER_S` and `TIMESTAMP` will triumph what's
-# auto-filled by Tune session.
-# TODO: Combine the following two with tune's, once there is a centralized
-#  file for both tune/train constants.
-TIMESTAMP = "timestamp"
-TIME_THIS_ITER_S = "time_this_iter_s"
```
Comment (justinvyu) [inline python/ray/train/constants.py:26]: Nice, cleaning this up!

File: python/ray/air/constants.py [hunk 2]
```diff
@@ -70,3 +70,32 @@
     "RAY_AIR_NEW_OUTPUT",
     "RAY_AIR_RICH_LAYOUT",
 }
+
+# The timestamp of when the result is generated.
+# Default to when the result is processed by tune.
+TIMESTAMP = "timestamp"
```
Comment (justinvyu) [inline python/ray/air/constants.py:76]: Nit: move this before the env variables section

File: python/ray/tune/result.py [hunk 3]
```diff
@@ -1,4 +1,9 @@
 import os
+from ray.air.constants import (
+    TIMESTAMP,
+    TIME_THIS_ITER_S,
+    TRAINING_ITERATION,
+)
```
Comment (justinvyu) [inline python/ray/tune/result.py:6]: Why do we need to import these again? Is it to preserve existing imports from `ray.tune.result`?

**Review by justinvyu — APPROVED** [review]
Thanks for cleaning this up.

Does it make sense to merge `ray.tune.result` with `ray.tune.constants`? No need to do in this PR.

File: python/ray/tune/result.py (hunk 3 above)
Comment (woshiyyya) [reply python/ray/tune/result.py:6]: There are other constant lists, such as `AUTO_RESULT_KEYS`, that still use these three constants. Once we migrate them together with the modules, there will be no need to import them again.

**Comment by woshiyyya** [comment]
> Does it make sense to merge ray.tune.result with ray.tune.constants?

@justinvyu yes that make sense. In the future the` ray.tune.result `should be removed, all constants go into `ray.tune.constants` and `ray.air.constants`.

**Review by krfricke — COMMENTED** [review]
LGTM but we should keep backwards compatibility.

Ideally we would warn users if they access the variables and guide them to the new import path. Do you think that's possible?

File: python/ray/tune/result.py [hunk 4]
```diff
@@ -144,17 +139,5 @@ def _get_defaults_results_dir() -> str:
...
-
-# Pickle File that stores parameters of the trial.
-EXPR_PARAM_PICKLE_FILE = "params.pkl"
-
-# File that stores the progress of the trial.
-EXPR_PROGRESS_FILE = "progress.csv"
-
-# File that stores results of the trial.
-EXPR_RESULT_FILE = "result.json"
-
```
[the remark this answers is not in the document]
Comment (woshiyyya) [reply python/ray/tune/result.py:158]: Good idea, it should make migrations safer:)

Done!

**Comment by woshiyyya** [comment]
@krfricke Yeah ideally we should warn people using another import path. Currently I can't think of a simple way to throw warnings only when importing a portion of constants in `tune/result.py`. I'll keep thinking about it. Do you have any idea.

**Comment by woshiyyya** [comment]
@krfricke One way I can think of is to do something like:

```python
# In ray/tune/results.py

from ray.air.constants import TRAINING_ITERATION

def import_warning(obj):
    warnings.warn("Importing from ray.tune.result is deprecated. Please import from ray.air.constants instead.", DeprecationWarning, stacklevel=3)
    return obj

TRAINING_ITERATION = import_warning(TRAINING_ITERATION)

OTHER_CONSTANTS = 123
```

So that only importing TRAINING_ITERATION will raise a warning. But yeah, it doesn't seems to be so elegant.

**Comment by krfricke** [comment]
> @krfricke One way I can think of is to do something like:
> 
> ```python
> # In ray/tune/results.py
> 
> from ray.air.constants import TRAINING_ITERATION
> 
> def import_warning(obj):
>     warnings.warn("Importing from ray.tune.result is deprecated. Please import from ray.air.constants instead.", DeprecationWarning, stacklevel=3)
>     return obj
> 
> TRAINING_ITERATION = import_warning(TRAINING_ITERATION)
> 
> OTHER_CONSTANTS = 123
> ```
> 
> So that only importing TRAINING_ITERATION will raise a warning. But, it doesn't seem to be that elegant.

Hm, but this would also raise a warning if only `OTHER_CONSTANTS` is imported, right?

**Comment by woshiyyya** [comment]
@krfricke Yes, this is a problem. I think raise deprecation warnings for classes is much easier than these constant variables. Currently I can't think of a good way to raise deprecation warnings only for a subset of constants. But given that these constants are mostly used by our developers, and for backward compatibility I also re-imported them in `ray/tune/result.py`, it should be fine for now?

Maybe we can merge this PR first, and come up with a better way later, so that it won't block the next result restoration PR to support Goku's tutorial?

**Review by krfricke — APPROVED** [review]
Yeah that's fine with me! Thanks!

**Pull request merged**